### PR TITLE
Add cloud provider inquiries to api-model

### PR DESCRIPTION
**Pull request by zvikorn**
Signed-off-by: Tzvika Avni <tavni@redhat.com>

https://issues.redhat.com/browse/SDA-4227

Tested with unformal integration tests.

### CODE REVIEW

File: model/clusters_mgmt/v1/available_regions_inquiry_resource.model [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Manages a specific available regions inquiry.
+resource AvailableRegionsInquiry {
+    // Retrieves the list of available regions of the cloud provider.
```
Comment (tzvatot) [inline model/clusters_mgmt/v1/available_regions_inquiry_resource.model:19]: indentation is wrong

**Comment by tzvatot** [comment]
LGTM

File: model/clusters_mgmt/v1/available_regions_inquiry_resource.model [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+limitations under the License.
+*/
+
+// Manages a specific available regions inquiry.
+resource AvailableRegionsInquiry {
+	// Retrieves the list of available regions of the cloud provider.
+	// IMPORTANT: This list doesn't currently support paging or searching, so the returned
+	// `page` will always be 1 and `size` and `total` will always be the total number of available regions
+	// of the provider.
+	method Search {
```
Comment (jhernand) [inline model/clusters_mgmt/v1/available_regions_inquiry_resource.model:23]: Are you sure you want to name this _Search_? That means that you want to use HTTP requests like this:

```http
POST /api/clusters_mgmt/v1/aws_inquiries/regions/search
````

If that is what you intended and paging isn't supported then I'd suggest to remove the `Page`, `Size` and `Total` parameters.

Comment (tzvatot) [reply model/clusters_mgmt/v1/available_regions_inquiry_resource.model:23]: @jhernand we need it as POST to allow payload. We wanted to keep the convention of Page/Size/Total although it's unsupported as parameters to this `search` endpoint, but for consistency of the API. If you think we can remove it, I'm ok with it.

Comment (jhernand) [reply model/clusters_mgmt/v1/available_regions_inquiry_resource.model:23]: I don't have any objection, just wanted to make sure that you intended to use that "/search" path segment.

File: model/clusters_mgmt/v1/gcp_inquiries_resource.model [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+limitations under the License.
+*/
+
+// Manages the collection of gcp inquiries.
+resource GCPInquiries {
+	// Reference to the resource that manages a collection of vpcs.
+	locator Vpcs {
+		target VpcsInquiry
+	}
+    // Reference to the resource that manages a collection of key rings.
```
Comment (tzvatot) [inline model/clusters_mgmt/v1/gcp_inquiries_resource.model:23]: indentation issue

File: model/clusters_mgmt/v1/gcp_inquiries_resource.model [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+	locator KeyRings {
+	    target KeyRingsInquiry
+	}
+    // Reference to the resource that manages a collection of encryption keys.
```
Comment (tzvatot) [inline model/clusters_mgmt/v1/gcp_inquiries_resource.model:27]: indentation issue

File: model/clusters_mgmt/v1/available_regions_inquiry_resource.model (hunk 2 above)
Comment (tzvatot) [reply model/clusters_mgmt/v1/available_regions_inquiry_resource.model:23]: My bad. According to the design doc, the endpoint should not contain "search" suffix. And given this implementation, it's indeed will not contain this suffix. Similar to /available_regions endpoint.

File: model/clusters_mgmt/v1/provider_data_inquiry_type.model [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Description of a cloud provider data used for cloud provider inquiries.
+struct CloudProviderData {
+  // Amazon Web Services settings.
```
Comment (tzvatot) [inline model/clusters_mgmt/v1/provider_data_inquiry_type.model:19]: indentation issue

File: model/clusters_mgmt/v1/aws_inquiries_resource.model [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Manages the collection of aws inquiries.
+resource AWSInquiries {
+
+  // Reference to the resource that manages a collection of vpcs.
+  locator Vpcs {
```
Comment (tzvatot) [inline model/clusters_mgmt/v1/aws_inquiries_resource.model:21]: indentation

**Pull request merged**